v2/models/field_type.go

```diff
@@ -0,0 +1,64 @@
+package models
+
+import "fmt"
+
+const GoNil = "nil"
+
+type PackageResolver interface {
```

Comment:
I defined the PackageResolver interface instead of using the package registry directly here to avoid cyclic import. I could not think of a better way to avoid it, but please let me know if you have any better ideas 🙇